rusk-recovery/src/state.rs

```diff
@@ -104,46 +106,43 @@ fn genesis_transfer(testnet: bool) -> TransferContract {
     transfer
 }
 
-const fn stake_amount(testnet: bool) -> Dusk {
-    match testnet {
-        true => dusk(2_000_000.0),
-        false => MINIMUM_STAKE,
-    }
-}
-
 /// Creates a new stake contract state with preset stakes added for the
 /// staking/consensus keys in the `keys/` folder. The stakes will all be the
 /// same and the minimum amount.
-fn genesis_stake(testnet: bool) -> StakeContract {
+fn genesis_stake(snapshot: &Snapshot) -> StakeContract {
     let theme = Theme::default();
     let mut stake_contract = StakeContract::default();
-
-    let stake_amount = stake_amount(testnet);
-
-    for provisioner in provisioners::keys(testnet).iter() {
-        let stake = Stake::with_eligibility(stake_amount, 0, 0);
+    snapshot.active_stakes().for_each(|staker| {
+        info!("{} Added provisioner", theme.action("Generating"),);
+        let stake = Stake::with_eligibility(
+            staker.amount,
+            staker.reward.unwrap_or(0),
+            staker.eligibility.unwrap_or(0),
+        );
         stake_contract
-            .insert_stake(*provisioner, stake)
+            .insert_stake(*staker.address(), stake)
             .expect("Genesis stake to be pushed to the stake");
         stake_contract
-            .insert_allowlist(*provisioner)
+            .insert_allowlist(*staker.address())
             .expect("Failed to allow genesis provisioner");
-    }
-    info!(
-        "{} Added {} provisioners",
-        theme.action("Generating"),
-        provisioners::keys(testnet).len()
-    );
+    });
+    snapshot.stake_owners().for_each(|provisioner| {
+        stake_contract
+            .add_owner(*provisioner)
+            .expect("Failed to set stake contract owner");
+    });
 
-    stake_contract
-        .add_owner(provisioners::owner(testnet))
-        .expect("Failed to set stake contract owner");
+    snapshot.allowlist().for_each(|provisioner| {
+        stake_contract
+            .insert_allowlist(*provisioner)
+            .expect("Failed to allow");
```

Comment:
There are too many expects for my taste. Consider that `expect` are just glorified `unwrap`s.
This can be okay temporary but we'll need to handle properly the error sooner rather than later.